docs/getting-started/data.md

```diff
@@ -0,0 +1,1202 @@
+
+## Big Ideas
+
+Gutenberg is part of WordPress core and frequently acts on the same data. There are posts, pages, taxonomies, widgets, navigation items and so on. The obvious way of using the data, would be to just request it from API whenever a given React component needs it. This would have a serious drawback, though. Any other component wanting to use the same data, would have to request it too. So now there is more than one request. Then, what if the data changes? Would all components re-request it? Would they even know?
+
+The data layer provides answers to all of these questions, and more. It handles data synchronization for you, so you can focus on your component. For example, when you need to do something with a post, you can access it with:
+
+```js
+function MyComponent({ widgetId }) {
+	const widget = useSelect(
+		select => select( coreStore ).getWidget( widgetId )
+	);
+
+	// ...
+}
+```
+
+And be sure that it’s the most recent version, and that no unnecessary HTTP requests were performed. How does it all work? There are a few big concepts to discuss:
+
+* Data package
+	* Selectors and resolvers 
+	* React hooks
+* Core-data package
+	* Entities
+	* Entity Records
+	* Data flow
+		* Reading Entity Records
+		* Editing Entity Records
+		* Saving Entity Records
+		* Deleting Entity Records
+
+# Data package
+## Selectors and resolvers
+Selectors are simple functions that return a piece of data from the redux store. Resolvers are used to load the data when there is none available yet. Let’s see how they work in tandem in this minimal store:
+
+```js
+const store = wp.data.createReduxStore( 'thermostat', {
+	// Essential functions
+	selectors: {
+		getTemperatureCelcius: ( state ) => state.temperature,
+		getTemperatureFarenheit: ( state ) => state.temperature * 1.8 + 32
+	},
+	resolvers: {
+		getTemperatureCelcius: () => ( { dispatch } ) => {
+			dispatch.receiveTemperature( 10 );
+		}
+	},
+
+	// Utility functions
+	__experimentalUseThunks: true,
+	actions: {
+		receiveTemperature: ( temperature ) => ({
+			type: 'RECEIVE_TEMPERATURE',
+			temperature
+		})
+	},
+	reducer(state={}, action) {
+		const newState = {
+			...state
+		}
+		if ( action.type === 'RECEIVE_TEMPERATURE' ) {
+			newState.temperature = action.temperature;
+		} else if ( action.type === '@@INIT' ) {
+			newState.temperature = 0;
+		}
+		return newState;
+	}
+} );
+wp.data.register(store)
+```
+
+The `@@INIT` action is dispatched when the store is instantiated, and so the initial state says `temperature: 0`
+
+### Simple selectors
+The  `getTemperatureFarenheit` is a simple selector, it predictably returns `0` once the store was instantiated:
+
+```js
+> wp.data.select('my-store').getTemperatureFarenheit()
+0
+```
+
+Note we didn’t provide the  `state` as an argument. The Gutenberg data layer handles that for us.
+
+### Memoized selectors
+
+dependency for other selectors
```

Comment:
I didn't understand why this sentence is here. Is this section complete?

Reply:
I think it's a placeholder I left to fill in later.